go/gilded-rose.go

```diff
@@ -8,51 +12,47 @@ type Item struct {
 func UpdateQuality(items []*Item) {
 	for i := 0; i < len(items); i++ {
 
-		if items[i].name != "Aged Brie" && items[i].name != "Backstage passes to a TAFKAL80ETC concert" {
-			if items[i].quality > 0 {
-				if items[i].name != "Sulfuras, Hand of Ragnaros" {
-					items[i].quality = items[i].quality - 1
-				}
+		if items[i].name == "Aged Brie" {
+			IncreaseQualityByOne(items[i])
+			if items[i].sellIn <= 0 {
+				IncreaseQualityByOne(items[i])
 			}
-		} else {
-			if items[i].quality < 50 {
-				items[i].quality = items[i].quality + 1
-				if items[i].name == "Backstage passes to a TAFKAL80ETC concert" {
-					if items[i].sellIn < 11 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
-					if items[i].sellIn < 6 {
-						if items[i].quality < 50 {
-							items[i].quality = items[i].quality + 1
-						}
-					}
+		} else if items[i].name == "Backstage passes to a TAFKAL80ETC concert" {
+			if items[i].sellIn > 0 {
+				IncreaseQualityByOne(items[i])
+				if items[i].sellIn < 6 {
+					IncreaseQualityByOne(items[i])
 				}
+				if items[i].sellIn < 11 {
+					IncreaseQualityByOne(items[i])
+				}
+
+			} else {
+				items[i].quality = 0
+			}
+
+		} else if strings.Contains(items[i].name, "Conjured") {
```

Comment:
Good that you are using strings here to account for other items outside of those already listed in the test cases. You could probably extend to use that for other item types but I dont think the spec gives enough info of item name formatting.